Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -178,5 +178,257 @@ public void DoGetRequest()
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Dataset.DatasetReference))]
+        [Alias("Dataset")]
+        public DatasetReference DefaultDataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The destination table to write the results into. If this is not specified, the 
+        /// results will be stored in a temporary table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Table.TableReference))]
+        [Alias("Table", "Dest")]
+        public TableReference DestinationTable { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Priority of the query.  Can be 'Batch' or 'Interactive'.
+        /// </para>
+        /// </summary>
+        public QueryPriority Priority { get; set; }
+
+        // Copy Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Copy.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoCopy)]
+        public SwitchParameter Copy { get; set; }
+
+        // Load Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Load.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoLoad)]
+        public SwitchParameter Load { get; set; }
+
+        // Extract Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Extract.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoExtract)]
+        public SwitchParameter Extract { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            var Client = BigQueryClient.Create(Project);
+
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.DoQuery:
+                    DoQuery(Client);
+                    break;
+                case ParameterSetNames.DoCopy:
+                    DoCopy(Client);
+                    break;
+                case ParameterSetNames.DoLoad:
+                    DoLoad(Client);
+                    break;
+                case ParameterSetNames.DoExtract:
+                    DoExtract(Client);
+                    break;
+                default:
+                    ThrowTerminatingError(new ErrorRecord(
+                        new Exception("You must select a valid BQ Job type."),
+                        "Type Not Found", ErrorCategory.ObjectNotFound, this));
+                    break;
+            }
```

Comment:
You can set a single variable from all of the functions above, and do a single `WriteObject` call here (if they return the same type of resource).